pkg/contracts/sui/gateway.go

```diff
@@ -57,22 +67,61 @@ func ActiveMessageContextDynamicFieldName() (json.RawMessage, error) {
 	return dynamicFieldNameToJSONArray("active_message_context")
 }
 
-// NewGatewayFromPairID creates a new Sui Gateway
-// from pair of `$packageID,$gatewayObjectID`
-func NewGatewayFromPairID(pair string) (*Gateway, error) {
-	packageID, gatewayObjectID, err := parsePair(pair)
+// NewGatewayFromAddress creates a new Sui Gateway struct from the gateway address in chain params.
+// The gateway address is in the form of `$packageID,$gatewayObjectID[,$withdrawCapID,originalPackageID]`
```

Comment:
gateway address is not a good name, imply a Sui address
pair is good
could be gatewayInfo as well